Lib/typing.py

```diff
@@ -2369,12 +2375,23 @@ def name_by_id(user_id: UserId) -> str:
         num = UserId(5) + 1     # type: int
     """
 
-    def new_type(x):
+    def __init__(self, name, tp):
+        self.__name__ = name
+        self.__qualname__ = name
+        self.__module__ = _callee(default='typing')
```

Comment:
That makes it rather slow but I guess `__module__` is actually helpful.

Reply:
It will be executed only once when `NewType` defined.  I think it should not dramatically effect performance.